rest/rest-api/src/main/java/org/apache/brooklyn/rest/domain/LocationSpec.java

```diff
@@ -53,7 +54,7 @@ public LocationSpec(
             @JsonProperty("config") @Nullable Map<String, ?> config) {
         this.name = name;
         this.spec = spec;
-        this.config = (config == null) ? Collections.<String, String> emptyMap() : ImmutableMap.copyOf(config);
+        this.config = (config == null) ? Collections.<String, String> emptyMap() : MutableMap.copyOf(config);
```

Comment:
(minor) could add `asUnmodifiable` or `asImmutableCopy` to keep the immutability.